runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -275,27 +300,127 @@ public void onGameStateChanged(GameStateChanged event)
 				taggedNpcs.clear();
 				break;
 			case LOGGED_IN:
-				if (config.amount() != -1
-					&& !config.taskName().isEmpty()
+				if (getStoredAmount() != -1
+					&& !getStoredTaskName().isEmpty()
 					&& loginFlag)
 				{
-					setExpeditiousChargeCount(config.expeditious());
-					setSlaughterChargeCount(config.slaughter());
-					setTask(config.taskName(), config.amount(), config.initialAmount(), config.taskLocation(), false);
+					setExpeditiousChargeCount(getStoredExpeditiousCharges());
+					setSlaughterChargeCount(getStoredSlaughterCharges());
+					setTask(getStoredTaskName(),
+						getStoredAmount(),
+						getStoredInitialAmount(),
+						getStoredTaskLocation(), false);
 					loginFlag = false;
 				}
 				break;
 		}
 	}
 
+	int getStoredAmount()
```

Comment:
naming these like getters/setters is somewhat confusing. Its probably clearest for them to be replaced with calls directly to configmanager

Reply:
I agree we could change the name, but I'm not sure replacing these with direct calls to the manager would be clearer. The call can get quite long to type out and the getters abstract away a null check to keep the behavior consistent with the old config setup.

Reply:
Just `getIntConfig(SlayerConfig.WHATEVER)` seems fine as it resolves the naming problem.

Reply:
Thank god for ctrl-alt-n